jupyterlab_git/handlers.py

```diff
@@ -391,6 +391,39 @@ async def post(self, path: str = ""):
         self.finish(json.dumps(output))
 
 
+class GitRemoteDetailsShowHandler(GitHandler):
+    """Handler for 'git remote -v'."""
+
+    @tornado.web.authenticated
+    async def get(self, path: str = ""):
+        """GET request handler to retrieve existing remotes."""
+        local_path = self.url2localpath(path)
+        output = await self.git.remote_show(local_path, verbose=True)
+        if output["code"] == 0:
+            self.set_status(201)
+        else:
+            self.set_status(500)
+        self.finish(json.dumps(output))
+
+
+class GitRemoteRemoveHandler(GitHandler):
+    """Handler for 'git remote remote <name>'."""
+
+    @tornado.web.authenticated
+    async def post(self, path: str = ""):
+        """GET request handler to remove a remote."""
+        local_path = self.url2localpath(path)
+        data = self.get_json_body()
+        name = data.get("name", None)
+
+        output = await self.git.remote_remove(local_path, name)
+        if output["code"] == 0:
+            self.set_status(201)
+        else:
+            self.set_status(500)
+        self.finish(json.dumps(output))
```

Comment:
Let's use delete as this is more appropriate. This means you cannot pass a body but you should include the _name_ in the endpoint `/git/<path>/remote/<name>`.

The status code should be 204.

```suggestion


class GitRemoteRemoveHandler(GitHandler):
    """Handler for 'git remote remote <name>'."""
    
    @tornado.web.authenticated
    async def delete(self, path: str = "", name: str = ""):
        """DELETE request handler to remove a remote."""
        local_path = self.url2localpath(path)

        output = await self.git.remote_remove(local_path, name)
        if output["code"] == 0:
            self.set_status(204)
        else:
            self.set_status(500)
        self.finish(json.dumps(output))
```